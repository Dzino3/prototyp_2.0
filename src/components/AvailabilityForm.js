import React, { useState } from 'react';
import './style/availabilityStyle.css'
import Axios from "axios"
import Availability from './components-info/Availability';

function AvailabilityForm({user}) {

  const [produktInfo, setProduktInfo] = useState([]);
  const [enteredBaseCode, setEnteredBaseCode] = useState("");


  const LookForProduct = () =>{
    Axios.post("https://codux.herokuapp.com/base_code", {base_code : enteredBaseCode}).then((response) =>{
        if (response.data.length !== 0){
          console.log(response.data)
            setProduktInfo(response.data[0]);
        }else{
            setProduktInfo([]);
        }
    });
    
};


  return (
    <>
    <form className='avalForm'>

    <div className='avalForm-inner'>
        
        <div className='avalForm-group'>
            <h2 className='avalH2'>Availability</h2>
            <label className='avalLabel'>Base Code:</label>
            <input className='avalInput' type="text" name='base_code' id='base_code' onChange={e => setEnteredBaseCode(e.target.value)} value={enteredBaseCode}></input>
            <input className='avalCheckButton' type="button" value="Check" onClick={LookForProduct}></input>
        </div>
        {produktInfo.length !== 0 ? <Availability produktInfo={produktInfo}></Availability> : <></>}
    </div>
    </form>
    </>
  )
}

export default AvailabilityForm
