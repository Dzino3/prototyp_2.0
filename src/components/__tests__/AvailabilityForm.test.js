import React from "react";
import renderer from 'react-test-renderer';
import AvailabilityForm from "../AvailabilityForm.js";

describe(`AvailabilityForm Rendering`, () => {

    it('should render AvailabilityForm with Availability', function () {
        let userX = {
            dealer: "Hallo42"
        };
        let tree = renderer.create(
            <AvailabilityForm user={userX}></AvailabilityForm>
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('should render AvailabilityForm with no Availability', function () {
        let userX = {}
        let tree = renderer.create(
            <AvailabilityForm user={userX}></AvailabilityForm>
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });
})