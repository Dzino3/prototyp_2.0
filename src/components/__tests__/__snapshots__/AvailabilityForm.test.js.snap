// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvailabilityForm Rendering should render AvailabilityForm with Availability 1`] = `
<form
  className="avalForm"
>
  <div
    className="avalForm-inner"
  >
    <div
      className="avalForm-group"
    >
      <h2
        className="avalH2"
      >
        Availability
      </h2>
      <label
        className="avalLabel"
      >
        Base Code:
      </label>
      <input
        className="avalInput"
        id="base_code"
        name="base_code"
        onChange={[Function]}
        type="text"
        value=""
      />
      <input
        className="avalCheckButton"
        onClick={[Function]}
        type="button"
        value="Check"
      />
    </div>
  </div>
</form>
`;

exports[`AvailabilityForm Rendering should render AvailabilityForm with no Availability 1`] = `
<form
  className="avalForm"
>
  <div
    className="avalForm-inner"
  >
    <div
      className="avalForm-group"
    >
      <h2
        className="avalH2"
      >
        Availability
      </h2>
      <label
        className="avalLabel"
      >
        Base Code:
      </label>
      <input
        className="avalInput"
        id="base_code"
        name="base_code"
        onChange={[Function]}
        type="text"
        value=""
      />
      <input
        className="avalCheckButton"
        onClick={[Function]}
        type="button"
        value="Check"
      />
    </div>
  </div>
</form>
`;
